src/core/dbus-cgroup.c

```diff
@@ -1966,6 +2008,61 @@ int bus_cgroup_set_property(
 
                 return 1;
         }
+        if (streq(name, "RestrictNetworkInterfaces")) {
+                int is_allow_list;
+                _cleanup_strv_free_ char **l = NULL;
+
+                r = sd_bus_message_enter_container(message, 'r', "bas");
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read(message, "b", &is_allow_list);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_read_strv(message, &l);
+                if (r < 0)
+                        return r;
+
+                r = sd_bus_message_exit_container(message);
+                if (r < 0)
+                        return r;
+
+                if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
+                        _cleanup_free_ char *joined = NULL;
+                        char **s;
+
+                        if (strv_isempty(l)) {
+                                c->restrict_network_interfaces_is_allow_list = false;
+                                c->restrict_network_interfaces = set_free(c->restrict_network_interfaces);
+
+                                unit_write_settingf(u, flags, name, "%s=", name);
+                                return 1;
+                        }
+
+                        if (!c->restrict_network_interfaces) {
+                                c->restrict_network_interfaces_is_allow_list = is_allow_list;
+                        }
+
+                        STRV_FOREACH(s, l) {
+                                if (c->restrict_network_interfaces_is_allow_list != (bool) is_allow_list)
+                                        set_remove(c->restrict_network_interfaces, *s);
```

Comment:
should be enclosed in free(), otherwise the entry is just removed but not freed

Reply:
I think I messed up with `string_hash_ops_free` vs `string_hash_ops_free_free`. I'm using `set_put_strdup` that uses `string_hash_ops_free`, so that's the reason why I have to call free, right?

Reply:
the destructors you define in the hash_ops are applied when the hashmap/set is destroyed as a whole, with the left-over items. but if you just use set_remove() you basically steal the items, they change into your possession, and are not implicitly freed.